carma_messenger_vehicle_plugin/src/move_over_law.py

```diff
@@ -34,6 +34,8 @@ def __init__(self, sumo_connector, veh_id):
         self.sumo_connector = sumo_connector
         self.sumo_connector.create_stop_veh(self._target_veh_id, self._stop_pos, self._stop_route)
         self.first_time_two_vehicles = True
+        self.is_get_closer = False
```

Comment:
Not advocating for any kind of change, but wanted to point out some thoughts for future cases like this: 

Generally at a design level, this is where something like a state machine would be good to consider. Right now you've got two variables `is_get_closer` and `is_stopped` (and possibly others here) that are really representing a single underlying state of the algorithm. The key is that these two variables are mutually exclusive, as far as I can tell, that the `is_get_closer` should never be true while `is_stopped` is true, and vice-versa. So they really encode an aspect of a single actual piece of information, having them as two separate variables like this doesn't really enforce or communicate that relationship between them.

A state machine implementation for something like this could be as simple as identifying the states of the algorithm and then defining an enum with a value per state:
```
class MoveOverLawState(Enum):
  INIT = 1
  GET_CLOSER = 2
  STOPPING = 3
  STOPPED = 4
  DEPART = 5
```

and then tracking it as a single `self.state` variable. This way you represent that there is really only one state value at a time (rather than two truly independent variables) and represented the total possible state space within that one variable. You could then design the class in a way that uses a switch statement or different methods per state, define transition conditions for each possible transition within the state space, etc. 
```
def process():
  if self.state == INIT:
    do_init()
    if transition_to_approaching(): 
      self.state = APPROACHING
  elif self.state == APPROACHING:
    do_approaching()
    if transition_to_stopping();
      self.state = STOPPING
  #etc, etc.
```

The value of that depends on how complex the design is, probably not necessary for this particular example.